arm-storsimple8000series/2017-06-01/swagger/storsimple.json

```diff
@@ -906,24 +999,80 @@
           }
         ],
         "responses": {
-          "202": {
-            "description": "Accepted the request to deactivate the device."
+          "200": {
+            "description": "The Alert settings.",
+            "schema": {
+              "$ref": "#/definitions/AlertSettings"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "put": {
+        "tags": [
+          "DeviceSettings"
+        ],
+        "description": "Creates or updates the alert settings of the specified device.",
+        "operationId": "DeviceSettings_CreateOrUpdateAlertSettings",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "deviceName",
```

Comment:
The four parameters subscriptionId, resourceGroupName, managerName, apiVersion, , are common across all operations. All other parameters are specific to APIs. Now, the current convention we followed for the order of parameters in all our APIs is, first all the non-ref. parameters are added, followed by the above mentioned ref. parameters. 
Your recommendation is also a good convention to follow, but the only hurdle I see is we already have these APIs deployed in production(Private preview). The post-private preview release is going now(which is equivalent to GA except for the # of customers asked to use), and we are nearing GA.  If we change the convention here, all the APIs will be impacted and will need changes on Portal(and thus another round of major testing) and BVT testcases.